service-statistics-pxweb/src/main/java/fi/nls/oskari/control/statistics/plugins/pxweb/PxwebStatisticalDatasourcePlugin.java

```diff
@@ -179,4 +185,27 @@ private String createUrl(String baseUrl, String pathId) {
         }
         return baseUrl;
     }
+
+    private ValueProcessor getValueProcessor(DatasourceLayer layer) {
+        if(layer.getConfig() == null) {
+            return DEFAULT_PROCESSOR;
+        }
+        String className = layer.getConfig("valueProcessor");
+        if(className == null) {
+            return DEFAULT_PROCESSOR;
+        }
+        ValueProcessor processor = processors.get(className);
+        if(processor != null) {
+            return processor;
+        }
+        try {
+            final Class clazz = Class.forName(className);
+            processor = (ValueProcessor) clazz.newInstance();
```

Comment:
Could add a `ValueProcessor.class.isAssignableFrom(clazz);` check here. The `Class.forName(String)` still needs to be inside the try-catch so doesn't make that much of a difference I guess.

Reply:
Added this check so misconfiguration now gives a more pleasant log-message without the stacktrace.